### PR TITLE
Added scrolling

### PR DESCRIPTION
**Pull request by bjedrzejewski**

### CODE REVIEW

File: src/assets/styles/scroll.less [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+.searchScroll .mCSB_scrollTools {
+  left: 2px;
+  margin-top: 10px;
+  margin-bottom: 10px;
+}
+
+.favScroll {
+  top: 52px;
+  bottom: 0px;
```
Comment (owennw) [inline src/assets/styles/scroll.less]: Is this 52px number calculated somehow, from pre-existing figures? It is also repeated -- I assume the places it's used are coupled?

File: src/assets/styles/variables.less [hunk 2]
```diff
@@ -18,4 +18,4 @@
 @expanded-width: 300px;
 
 // Heights
-@splitter-height: 2px;
\ No newline at end of file
+@splitter-height: 1px;
```
Comment (owennw) [inline src/assets/styles/variables.less]: Why has this changed?

File: src/assets/styles/scroll.less (hunk 1 above)
Comment (owennw) [inline src/assets/styles/scroll.less]: In fact, search scroll and favourite scroll look to be identical -- can't we just use one class?

File: src/assets/styles/variables.less (hunk 2 above)
Comment (bjedrzejewski) [inline src/assets/styles/variables.less]: to look more like the design

File: src/assets/styles/scroll.less (hunk 1 above)
Comment (bjedrzejewski) [inline src/assets/styles/scroll.less]: at the moment they are identical, we need 2 classes to target them with javascript separately. could add 3rd class for styling. or have class for styling target with ids

File: src/index.html [hunk 3]
```diff
@@ -57,6 +62,16 @@
...
+    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
+    <script src="http://cdn.jsdelivr.net/jquery.mcustomscrollbar/3.0.6/jquery.mCustomScrollbar.concat.min.js"></script>
+    <script>
+        (function($){
+            $(window).load(function(){
+                $('.searchScroll').mCustomScrollbar();
+                $('.favScroll').mCustomScrollbar();
+            });
+        })(jQuery);
+    </script>
```
Comment (owennw) [inline src/index.html]: I take it we have to use jQuery? I think I'd sooner see this script in a separate js file, feels neater. Do you agree?

File: src/assets/styles/favourites.less [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ﻿@favourite-height: 150px;
-@favourite-padding-left: 0px;
+@favourite-padding-left: 20px;
 @favourite-padding-top: 5px;
 @favourite-padding-right: 5px;
```
Comment (owennw) [inline src/assets/styles/favourites.less]: There isn't a corresponding change for search.less?

File: src/assets/styles/variables.less (hunk 2 above)
Comment (owennw) [inline src/assets/styles/variables.less]: The design did specify a 2px thick splitter (at least, that's what Rui said before)

**Comment by bjedrzejewski** [comment]
Build failed randomly, will look into that

File: src/assets/styles/scroll.less (hunk 1 above)
Comment (owennw) [inline src/assets/styles/scroll.less]: If jQuery needs two classes that's okay, but the css can still be tidied, e.g. use `.favScroll, .searchScroll` in the two duplicated places.

File: src/assets/styles/favourites.less (hunk 4 above)
Comment (bjedrzejewski) [inline src/assets/styles/favourites.less]: Search already is padded out by 20px on details

File: src/assets/styles/scroll.less (hunk 1 above)
Comment (bjedrzejewski) [inline src/assets/styles/scroll.less]: Yeah, I refactored, used 2 separate ids and one class. Originally these were different.

File: src/assets/styles/variables.less (hunk 2 above)
Comment (bjedrzejewski) [inline src/assets/styles/variables.less]: Ok, will revert then. His design definitely has 1px splitter though (on the pdfs)

File: src/index.html (hunk 3 above)
Comment (bjedrzejewski) [inline src/index.html]: Yeah, we need to use jQuery. There was never any hate for jQuery when jQuery is needed. Colin (and people) just warn others not to use jQuerry for whats Angular job.

Comment (bjedrzejewski) [inline src/index.html]: And yeah, will move the script now

File: src/assets/styles/style.less [hunk 5]
```diff
@@ -8,6 +8,7 @@
 @import "favourites.less";
 @import "star.less";
 @import "showcase-changes.less";
+//@import "custom-scroll.less";
```
Comment (owennw) [inline src/assets/styles/style.less]: I don't think we want this?

Comment (bjedrzejewski) [inline src/assets/styles/style.less]: i just removed that

Comment (bjedrzejewski) [inline src/assets/styles/style.less]: i did it when trying to fix that issue

**Comment by owennw** [comment]
scroll_bug
1. I think moving the chart 20px right isn't good enough, as now the numbers if big enough go through the tile as shown. 

2. When the search bar is open, clicking on the far left of the favourites tab won't select it, as that's where the hidden scroll bar is that stops the bar being selected.

3. I can't see the design because I was kicked from the conversation.. but I'm not sure if the tile splitters should be visible to the left of the scroll bars?

**Comment by bjedrzejewski** [comment]
1. Ok, will fix it, just padding on the title.
2. This is not an issue.
3. Thats what the design shows

**Comment by bjedrzejewski** [comment]
@owennw thanks for thorough review, the latest version should be ok

**Comment by owennw** [comment]
Just two more things:

1. Should the cursor on the scroll bar still be the hand? I would have thought the default pointer was more usual. (Perhaps a question for a UX person?)
2. On both of the tabs when the scrollbar is present use the scroll wheel with the mouse over the centre of the tab to scroll through the results. Notice that a) the highlighted result does not change and b) when the scrolling stops the highlighted row is only updated when the mouse is moved slightly.

I don't know how large an issue (2) is, so might be best spun off as a different issue?

**Comment by bjedrzejewski** [comment]
1. I actually prefer the hand here, we can later raise as an ux question.
2. This is difficult to fix, so I would make it another issue if we want to fix it. Somehow we need to fire some events on mouse scroll?

**Comment by owennw** [comment]
Okay can you raise those two issues?
